README.txt
README

This file is still draft, maybe some day it will be filled with some interesting info.

Always look for *.example files, these contains specific information about additional files used by generator

If you have unpacked all the content to one directory, just browse to the install location of your game.

And please, read manual!

To install MySQL database >= 4.1 you should use UTF-8 charset format! Unfortunately it would be breaking the standard of MySQL data files so you in that case the database should be reinstalled and there will be no compatibility with MySQL < 4.1 data files. You can also use latin1 or latin2 charset (in that case data files will be compatible) but if any illegal character found SQL query it will be ignored... ;(

If you use MySQL < 4.1 backups can be made directly by copying data files from MySQL datadir (/var/lib/mysql/ or "C:\Program Files\MySQL\data\").

If running GF on a machine < 1 GB RAM it is recommended to set output_buffering value in your PHP parser to 16KB (16384) or even 64KB (65536). If your machine is dedicated web server you can also turn on zlib compression (useful for sending data to the world). 

Here is my example from home server:

output_buffering=32768
zlib.output_compression=On


CHANGES.txt
2010-02-07	[ 0.5.3 ]

Update: Graphics conversion to true color PNG format, do not support GIF any more
Update: HTML style optimalization


2010-01-15	[ 0.5.2 ]

Update: Graphics and artwork
Update: Javascript library update
Bug: CSS and HTML fixes for correct display on Safari
New: Support for "box" section of modules, output is generated by style_module_section()
New: Handler support - now used only by style_box_head() and style_box_foot() which will be default instead tablebegin() and tableend() which are defined in galaxy (forces) output style only
New: Bug module for developers (limited to wheel group by default)
New: Default style supports sections "top" and "bottom" from now
Change: Galaxy game engine parameters now may be defined directly in site configuration without changing code


2010-01-03	[ 0.5.1 ]

Update: Newer MySQL driver for use UTF-8 locales
Change: All PHP scripts are not using short tags anymore
Change: All text files now have .txt extension
Bug: Fixes types and checks are now logged into security.log
Add: Advertisement mechanism from Stellar Quest
Update: Function str_sqlsafe() added again to provide compatilibity 
New: Module "censorship" inspired by Stellar Quest's autocensoring mechanism
New: Module "rank" from Stellar Quest "include/common.php"
New: Site menu is now configured in config.php and handled by style_menu_galaxy()


2009-12-31	[ 0.4.7 ]

After very long time original team is going to work under new relase of Galaxy Forces project. 
Stay tuned!


2005-11-19	[ 0.4.6 ]

Bug fixed: Possibility in javascript execution in bbcode - thx to sharkpp
Bug fixed: Changing colony description - thx to lander
Change: Antispam option - thx to sharkpp
Change: Little change in code - another few functions went away.
New: Colony crew members...


2005-11-12	[ 0.4.5 ]

New: Site maintenance option
New: Options in administration panel by unk
New: Droping, giving items
Change: User authentication method updated
Change: Small database update, some code may not work properly :(
Bug fixed: Researches can be finished in one turn - thx to undeath


2005-11-02	[ 0.4.4 ]

Bug fixed: Attacks
Bug fixed: Credit cards - thx lander
New: Robot AX-3, Robot AX-6, Walker, Valkyrie units available for human technology
Change: New round counter - thx unk
Change: Sorting by IP - thx unk
Change: Colony building amount in build list - thx jaco!!!


2005-10-25	[ 0.4.3 ]
	
New: Referer blocking module (testing)
New: Research level
Bug fixed: Hawk amount in attacks missed - thx to carrigan
Changes: Research may be cancelled now


2005-10-21	[ 0.4.2 ]

Bug fixed: Crystals can be bought in markets again
Bug fixed: Notices when donating clan
Bug fixed: When production goes too long you wont see percentage
Bug fixed: Defense reports are more true now
New: Crystals can be eaten by local lifeform (depend on planet life signs)
New: 10% for communication lost when trying to cancel attack
New: BX-1 robots now can be damaged


2005-10-19	[ 0.4.1 ]

New: Drinks are now ready to use (drink ;)
New: Descriptions for all units
New: Emoticons in messages
Bug fixed: Case insentitive search when changing clan status
Bug fixed: Attack durations now less but can be more than 6 turns
Bug fixed: Attack/Defense reports changed
Changes: Level up only by 1 level
Changes: More resources can be found by exploration
Changes: More experience in mines


2005-10-15	[ 0.4.0 ]

Changes: Lots of bugfixes and small changes in the code...
Changes: Attacks costs credits and more resources but are a bit (6 times) faster
Changes: Whisper unit, ported from SSE - thanks to shark++ and jaco!!!
Changes: Sources are not unlimited from now - thanks to aydamusek
Bug fixed: New PM module - thanks to login_error


2005-10-11	[ 0.3.17 ]

Changes: Some updates... No time for log...
Changes: GF Crew updated ?-)
Changes: Sources now available at sourceforge.net


2005-10-03	[ 0.3.16 ]

Brought GF into life again. No time for log...


2005-06-27	[ 0.3.15 ]

Bug fixed: Chat now works from Links browser
Bug fixed: Chat messages are now being deleted correctly!
Bug fixed: Resources not shown if there are only sources (without any silos)
Bug fixed: Wrong page count on messages page
Bug fixed: Avatar change in colony and finger pages
Bug fixed: Fixes in messages page
Bug fixed: Colony corruption if no silos reported as to less energy to keep silos on - fixed report
New: Colony description
New: Password change for administrators
New: Chat administration for moderators
New: RSS channels for chat and news
New: Planets in Wolf galaxy
New: Planet scanning (mp cost depends on scanner type)
New: Selling items in item shops
New: Setup for installing GF locally :-)
New: BB for private messages
Changes: Less score from working in mines
Changes: Teleports can provide more than one target
Changes: Markets can sell everything now (even energy)
Changes: Chat module fixed
Changes: You will loose 15% score if colony abandoned
Changes: Reputation affects on prices in item/gem shops and markets
Changes: Default theme improved to be better viewed with Konqueror browser
Changes: Project tree rebuild a bit


2005-05-13	[ 0.3.13 ]

Changes: HTML links improved to make Opera always refresh content.
New: Colony details at whois.
New: New style editor for sending messages.
New: Clan administration at finger (whois)


2005-05-11	[ 0.3.12 ]

New: lot of bugs :-)
New: Experience system changed but not finished so locked now!
New: Tron technology.


2005-04-10	[ 0.3.11 ]

Changes: Base level 5 has now less requirements.
Changes: Corruption if uran silos overloaded.
Changes: Detectors are less effective but can be supported with satellites.
Changes: Sounds can be turned off in profile.
Changes: Only one database connection attempt.
Changes: Maximum 32 chars for username.

ads.php
<?php

// ===========================================================================
// Ads {ads.php}
// ===========================================================================

// ---------------------------------------------------------------------------
//	Version:	1.2
//	Modified:	2007-06-20
// ---------------------------------------------------------------------------

// ===========================================================================
// This file is a part of Galaxy Forces SSE.
// ===========================================================================

$index = 'ads';
$auth = true;

// JavaScript code
$js[] = 'functions';
$js[] = 'e107';

require('include/header.php');

$ad = getvar('ad');
$view = getvar('view');
$login = strtolower($login);

locale('content/ads');

$pagename = $Lang['Ads'];

// ===========================================================================
// ACTION
// ===========================================================================

if ($action == 'add') {
	list($lang, $type, $title, $message, $time, $status, $notify) = array(getvar('lang'), getvar('type'), str_sqlsafe(htmlspecialchars(strip_tags(getvar('title')))), str_sqlsafe(htmlspecialchars(str_replace("\n", '[br]', (strip_tags(getvar('message')))))), getvar('time'), getvar('status'), getvar('notify'));
	if (!$title || !$message || ($Player['usergroup'] != 'admin' && !$time)) $errors = $Lang['NoInput'].'<br />';
	else {
		if (function_exists('censorship') && !censorship($message)) $errors = $Lang['BackOff'].'<br />';
		if ($Player['usergroup'] != 'admin') $status = '0';
		if ($notify) $notify = '1';
		$time = 6 * ceil($time / 6); if ($time < 6) $time = 6; else if ($time > 72) $time = 72;
		$expires = $time * 3600 + time();
		$class = $Player['usergroup'] ? $Player['usergroup'] : 'user';
echo "INSERT INTO `${prefix}ads` (`expires`,`time`,`lang`,`class`,`author`,`title`,`type`,`content`,`status`,`notify`) VALUES ('".$expires."','".time()."','".$lang."','".$class."','".$Player['login']."','".$title."','".$type."','".$message."','$status','$notify');";		
		if ($db->query("INSERT INTO `${prefix}ads` (`expires`,`time`,`lang`,`class`,`author`,`title`,`type`,`content`,`status`,`notify`) VALUES ('".$expires."','".time()."','".$lang."','".$class."','".$Player['login']."','".$title."','".$type."','".$message."','$status','$notify');")) $result .= $Lang['AdAdded'].'<br />';
	else $errors = mysql_error().'<br />';
	}
}
else if ($action == 'reply') {
	$ad = getvar('ad');
	$message = str_sqlsafe(htmlspecialchars(str_replace("\n", '[br]', (strip_tags(getvar('message'))))));
	if (!$message) $errors = $Lang['NoInput'].'<br />';
	else if ($db->query("SELECT id,status,notify,author,title FROM `${prefix}ads` WHERE id='$ad' AND type!='reply' LIMIT 1;") && (!$t = $db->fetchrow())) $errors = $Lang['AdNotExists'].'<br />';
	else if ($t['status'] != '2') {
		if (function_exists('censorship') && !censorship($message)) $errors = $Lang['BackOff'].'<br />';
		if ($t['author'] != $login && $t['notify'] == '1') sendmessage($Lang['TitleNotify'], '<a href="whois.php?player='.$login.'">'.strcap($login).'</a> '.$Lang['MsgNotify'].' <a href="ads.php?ad='.$ad.'" class="result">"'.$t['title'].'"</a>', 'Robot', $t['author']);
		$class = $Player['usergroup'] ? $Player['usergroup'] : 'user';
		if ($db->query("INSERT INTO `${prefix}ads` (`class`,`time`,`author`,`title`,`type`,`content`) VALUES ('".$class."','".time()."','".$Player['login']."','".$ad."','reply','".$message."');")) $result .= $Lang['ReplySent'].'<br />';
		else $errors = mysql_error().'<br />';
		$db->query("UPDATE `${prefix}ads` SET replies=replies+1 WHERE id='$ad';");
	} else $errors = 'Whoops! <br />';
}
else if ($action == 'edit') {
	$ad = getvar('ad');
	list($lang, $type, $title, $message, $time, $status, $notify) = array(getvar('lang'), getvar('type'), str_sqlsafe(htmlspecialchars(strip_tags(getvar('title')))), str_sqlsafe(htmlspecialchars(str_replace("\n", '[br]', (strip_tags(getvar('message')))))), getvar('time'), getvar('status'), getvar('notify'));
	if ($db->query("SELECT type,title FROM `${prefix}ads` WHERE id='$ad' LIMIT 1;") && ($t = $db->fetchrow())) {
		if (((!$message || !$title) && $t['type'] != 'reply') || (!$message)) $errors .= $Lang['ErrorNoContent'].'<br />';
		else {
			if (function_exists('censorship') && !censorship($message)) $errors = $Lang['BackOff'].'<br />';
			$notify = ($notify ? $notify :'');
			$clausule = ($Player['usergroup'] == 'admin' ? '' : " AND `author`='".$login."'");
			if ($Player['usergroup'] != 'admin') $status = '0';

			if ($t['type'] == 'reply') {
				$type = 'reply';
				$time = time();
				$title = $t['title'];
			}
			
			$time_sql = '';
			if ($time) {
				$time = 6 * ceil($time / 6);
				if ($time < 6) $time = 6;
				else if ($time > 72) $time = 72;
				
				$time_sql = ",`expires`='".( time() + $time * 3600 )."'";
			}
			if ($db->query("UPDATE `${prefix}ads` SET lang='$lang',title='$title',type='$type',content='$message',status='$status',notify='$notify'".$time_sql." WHERE `id`='".$ad."'".$clausule." ;")) $result = $Lang['PostEdited']."<br />";
			else $errors = mysql_error().'<br />';
		}
	} else $errors = $Lang['AdNotExists'].'<br />';
}
else if ($action == 'remove') {
	$ad = getvar('ad');
	if ($db->query("SELECT `type`,`title`,`author` FROM `${prefix}ads` WHERE id='$ad' LIMIT 1;") && $t = $db->fetchrow()) {
		if ($Player['usergroup'] == 'admin' || $Player['usergroup'] == 'mod' || $Player['usergroup'] == 'global_mod') $clausule = '';
		else $clausule = " AND `author`='".$login."'";
		if ($db->query("DELETE FROM `${prefix}ads` WHERE id='$ad' OR (type='reply' AND title='$ad') ".$clausule.";")) {
			if ($t['type'] == 'reply') {
				$ad = $t['title'];
				$db->query("UPDATE `${prefix}ads` SET replies=replies-1 WHERE id='$ad';");
			}
			if ($t['type'] == 'reply') {
				$result = $Lang['ReplyDeleted'].'<br />';
				if ($t['author'] != $login) sendmessage($Lang['TitleReDeleted'], '<a href="whois.php?player='.$login.'">'.strcap($login).'</a> '.$Lang['MsgReDeleted'].' <a href="ads.php?ad='.$t['title'].'" class="result">"'.$t['title'].'"</a>', 'Robot', $t['author']);
			}
			else {
				$result = $Lang['AdDeleted'].'<br />';
				if ($t['author'] != $login) sendmessage($Lang['TitleAdDeleted'], '<a href="whois.php?player='.$login.'">'.strcap($login).'</a> '.$Lang['MsgAdDeleted'].' <span class="result">"'.$t['title'].'"</span>', 'Robot', $t['author']);
			}
		}
		else $errors = $Lang['Error'].' '.mysql_error().'<br />';
	}
	else $errors = $Lang['AdNotExists'].'<br />';
}

// ===========================================================================
// ERROR
// ===========================================================================

if ($errors) {
	tablebegin("<font class=\"error\">${Lang['Error']}!</font>", '400');
	echo "\t\t<h3>${Lang['ErrorProblems']}</h3><font class=\"error\">$errors</font><br />";
	echo "<a href=\"javascript:history.back(1)\">${Lang['GoBack']}&nbsp;&gt;&gt;</a><br /><br />";
	tableend($pagename);
}

// ===========================================================================
// RESULTS
// ===========================================================================

elseif ($result) {
	tablebegin($pagename, 400);
	echo "\t\t<br /><font class=\"result\">$result</font><br /><a href=\"${_SERVER['PHP_SELF']}\">${Lang['GoBack']}&nbsp;&gt;&gt;</a><br /><br />";
	tableend($pagename);
}

// ===========================================================================
// ADS LIST
// ===========================================================================

else if (!$view) {
	tablebegin($pagename);
		echo "<br /><span class=\"h3\">$pagename</span><br /><br />";
		echo "<input type='hidden' name='idads' id='idads' value='0' /><table id=\"sub\" cellspacing=\"0\" cellpadding=\"0\">\n\t".'
			<tr id="header">
				<td width="12" id="headerl">&nbsp;</td>
				<td width="50">
					<a href="" class="result">'.$Lang['Language'].'</a>
				</td><td width="8">&nbsp;</td><td width="75">
					<a href="" class="capacity">'.$Lang['Category'].'</a>
				</td><td width="8">&nbsp;</td><td width="75">
					<a href="">'.$Lang['Author'].'</a>
				</td><td width="8">&nbsp;</td><td align="left" width="40%">
					<a href="" class="result">'.$Lang['Title'].'</a>
				</td><td width="8">&nbsp;</td><td width="50">
					<a href="">'.$Lang['Replies'].'</a>
				</td><td width="8">&nbsp;</td><td width="50">
					<a href="">'.$Lang['Time'].'</a>
				</td><td width="8">&nbsp;</td><td width="50">
					<a href="" class="minus">'.$Lang['Expires'].'</a>
				</td>
				<td width="12" id="headerr">&nbsp;</td>
			</tr>';
			
		
		// Read from db, remove expired msgs
		
		$tab = array();
		$del = false;

		$db->query("SELECT * FROM `${prefix}ads` WHERE `type` != 'reply' AND `expired`=0;");
		while ($t = $db->fetchrow()) {
			if ($t['expires'] <= $now) $del=true;
			else $tab[] = $t;
		}
		if ($del) $db->query("UPDATE `${prefix}ads` SET `expired`=1 WHERE `expires`<=".time());
?>
	<script>
	<!--

	function ask($url) {
		if (confirm('<?php echo $Lang['AreYouSure?']; ?>')) location.href = $url;
	}

	//-->
	</script>
<?php
		if (count($tab)) foreach ($tab as $t) {
			$i++;
			if (isset($Ranks[$t['class']]['chat'])) $id = ' style="'.$Ranks[$t['class']]['chat'].'"';
			else $id = (! ($i % 2)) ? $id = ' id="div"' :  $id = '';
			// Generate ads list
			echo '<tr align="center"'.$id.'>
					<td>&nbsp;</td><td class="result">
						'.@$Lang['Lang[]'][$t['lang']].'
					</td><td>&nbsp;</td><td class="capacity">
						'.@$Lang['AdTypes[]'][$t['type']].'
					</td><td>&nbsp;</td><td>
						<a href="whois.php?player='.$t['author'].'">'.strcap($t['author']).'</a>
					</td><td>&nbsp;</td><td align="left">
						<a href="?ad='.$t['id'].($page ? "&page=$page" : '').'" class="result">'.$t['title'].' &raquo;</a>
					</td><td>&nbsp;</td><td>
						<a class="onclick" onclick=\'ADS.replies("'.$t['id'].'")\'>'.$t['replies'].' &raquo;</a>
					</td><td>&nbsp;</td><td>
						'.($t['time'] ? date('Y-m-d H:i ', (int)$t['time']) : $Lang['Unknown']).'
					</td><td>&nbsp;</td><td class="minus">
						'.($t['expires'] ? date('Y-m-d H:i', (int)$t['expires']) : $Lang['Never']).'
					</td><td>&nbsp;</td>';
				echo '</tr>'; //href="?ad='.$t['id'].($page ? "&page=$page" : '').'#replies"
				echo "<tr style='height: 0px; padding: 0;'><td style='line-height: 0px; padding: 0;' colspan='15'><center><div style='width:500px;' id='ads{$t['id']}'></div></center></td></tr>";
			// View ad
			if (isset($ad) && $ad == $t['id']) {
				require('include/bbcode.php');
				echo '<tr align="center">
						<td colspan="15">
							<br /><table cellspacing="0" cellpadding="0" style="width: 75%;">
								<tr id="header">
									<td id="headerl">&nbsp;</td><td>'.$Lang['Message'].'</td><td id="headerr">&nbsp;</td>
								</tr><tr id="div">
									<td id="headerl">&nbsp;</td><td>'.bbcode($t['content']).'<br /><div align="right"><b><a href="?ad='.$ad.'&view=reply">'.$Lang['Reply'].' &raquo;</a></b>';
				if ($t['author'] == $login || in_array($Player['usergroup'], array('admin', 'global_mod'))) echo " &nbsp; <a href=\"?ad=$ad&view=edit\">${Lang['Edit']} &raquo;</a> &nbsp;  <a href=\"javascript:ask('ads.php?action=remove&ad=".$t['id']."')\" class=\"minus\">${Lang['Delete']} &raquo;</a>";
				echo '</div></td><td id="headerr">&nbsp;</td>
								</tr><tr>
									<td id="headerb" colspan="3">&nbsp;</td>
								</tr>
							</table>';
				// Ad have replies?
				if ($t['replies'] > 0) {
					$db->query("SELECT * FROM `${prefix}ads` WHERE type = 'reply' AND title = $ad;");
					
					echo '<a name="replies"></a><table cellspacing="0" cellpadding="0" style="width: 50%;">
							<tr id="header">
								<td id="headerl">&nbsp;</td><td>'.$Lang['Replies'].'</td><td id="headerr">&nbsp;</td>
							</tr>';
					while ($r = $db->fetchrow()) {
						if (isset($Ranks[$r['class']]['chat'])) $id = ' style="'.$Ranks[$r['class']]['chat'].'"';
						else $id = ' id="div"';
						echo '<tr'.$id.'><td id="headerl">&nbsp;</td><td>'.bbcode($r['content']).'<br /><b><a href="whois.php?player='.$r['author'].'">'.strcap($r['author']).'</a> at <span class="minus">'.date('Y-m-d H:i', $r['time']).'</span></b>';
						if ($t['author'] == $login || in_array($Player['usergroup'], array('admin', 'global_mod'))) echo "<div align=\"right\"><a href=\"?ad=".$r['id']."&view=edit\">${Lang['Edit']} &raquo;</a> &nbsp;  <a href=\"javascript:ask('ads.php?action=remove&ad=".$r['id']."')\" class=\"minus\">${Lang['Delete']} &raquo;</a></div>";
						echo '</td><td id="headerr">&nbsp;</td>
							
							</tr>';
					}
					if (!$db->numrows()) echo '<tr id="headerbox" align="center"><td id="headerl">&nbsp;</td><td><span class="minus">'.$Lang['NoReplies'].'</span></td><td id="headerr">&nbsp;</td></tr>';
					echo '<tr>
								<td id="headerb" colspan="3">&nbsp;</td>
							</tr>
						</table>
						</td>
				<tr>';
				}
			}			
		}
		
		if (mysql_error()) echo '<tr align="center"><td colspan="15"><br /><span class="minus">'.mysql_error().'</span></td><tr>'; // whoops!
		
		if (!count($tab)) echo '<tr align="center"><td colspan="15"><br /><span class="minus">'.$Lang['NoAds'].'</span></td><tr>'; // no ads :(

		echo '</table>';
		
		tablebreak();
	
		echo '<br /><a href="?view=add">'.$Lang['AddAd'].' &raquo;</a><br /><br />';
	tableend($pagename);
}

// ===========================================================================
// ADD / EDIT
// ===========================================================================

else if ($view == 'add' || $view == 'edit' || $view == 'reply') {
	$expires = $title = $lang = $message = $expires = '';
	if (!$lang) $lang = $Player['language'];

	switch($view) {
		case 'add': $pagename = $Lang['AddAd']; $submit = $Lang['Save']; break;
		case 'edit': $pagename = $Lang['EditAd']; $submit = $Lang['Edit']; break;
		case 'reply': $pagename = $Lang['Reply'];  $submit = $Lang['Reply']; break;
	}

	if ($view == 'edit') {
		$ad = getvar('ad');
		$db->query("SELECT `expires`,`lang`,`title`,`content`,`type`,`status`,`notify` FROM `${prefix}ads` WHERE id='$ad' LIMIT 1;");
		if ($row = $db->fetchrow()) {
			$expires = $row['expires'];
			$title = $row['title'];
			$lang = $row['lang'];
			$type = $row['type'];
			$message = $row['content'];
			$status = $row['status'];
			$notify = $row['notify'];
		} else echo mysql_error().'No!';
		
	}

	tablebegin($pagename, 500);

	echo '<br /><span class="h3">'.$pagename.'</span><br /><br />';
	echo '<form action="ads.php" method="post" name="ads"><input type="hidden" name="action" value="'.$view.'"><input type="hidden" name="ad" value="'.$ad.'">
		<table id="form" align="center" cellspacing="0" cellpadding="0" border="0">';
	if (($view !== 'edit' || @$type !== 'reply') && $view !== 'reply') { echo '
		<tr><td colspan="3">&nbsp;</td></tr>
		<tr>
			<td><b>'.$Lang['Title'].':</b></td>
			<td>&nbsp;</td>
			<td><input type="text" size="55" maxlength="40" name="title" value="'.$title.'" />
				<select name="type">';
	 
	foreach ($Lang['AdTypes[]'] as $key => $v) echo '<option value="'.$key.'"'.($key == $type ? ' selected' : '').'>'.$Lang['AdTypes[]'][$key].'</option>';
	echo '</select></td>
		</tr>
		<tr>
			<td><b>'.$Lang['Time'].':</b></td>
			<td>&nbsp;</td>
			<td><select name="time">'.($expires ? '<option value="0">'.$Lang['DoNotChange'].'</option>' : '' );
			for ($i = 24; $i <= 168; $i += 24) { echo '<option value="'.$i.'">'.$i.' '.$Lang['hours'].'</option>'; }
		echo '</select></td>
		</tr>
		<tr>
			<td><b>'.$Lang['Language'].':</b></td>
			<td>&nbsp;</td>
			<td><select name="lang">';
			foreach ($Lang['Lang[]'] as $key => $v) echo '<option value="'.$key.'"'.($key == $lang ? ' selected' : '').'>'.$Lang['Lang[]'][$key].($key ? " ($key)" : '').'</option>';
		echo '</tr>';
		if ($Player['usergroup'] == 'admin') echo '
		<tr>
			<td><b>'.$Lang['Status'].':</b></td>
			<td>&nbsp;</td>
			<td><select name="status">
				<option value="0"'.($status == 0 ? ' selected' : '').'>'.$Lang['Status[]'][0].'</option><option value="1"'.($status == 1 ? ' selected' : '').'>'.$Lang['Status[]'][1].'</option><option value="2"'.($status == 2 ? ' selected' : '').'>'.$Lang['Status[]'][2].'</option>
			</tr>';
		echo '<tr>
			<td colspan="2">&nbsp;</td>
			<td><input type="checkbox" name="notify"'.(@$notify ? ' checked' : '').'> '.$Lang['NotifyAnsvers'].'</td>
			</tr>
			<tr><td colspan="3">&nbsp;</td></tr>';
		}
		echo '
		<tr>
			<td><b>'.$Lang['Message:'].'</b></td>
			<td>&nbsp;</td>
			<td>';
	require('include/editor.php');
	editor('message', 72, 20, 'left', str_ireplace( '[br]', '', $message ) );
	echo '</td>
		</tr>
		<tr><td colspan="3">&nbsp;</td></tr>
		<tr>
			<td colspan="3">';

	echo '<center><input type="submit" value="'.$submit.'" /></center>';
?>
	</td>
	</tr>
	<tr><td colspan="3">&nbsp;</td></tr>
	</table>
<?php
			
			echo '</form>';
	tableend("<a href=\"ads.php\">${Lang['GoBack']} &raquo;</a>");
} 

require('include/footer.php');
